src/plugins/discover/public/application/main/components/sidebar/discover_field.tsx

```diff
@@ -8,162 +8,70 @@
-          content={i18n.translate('discover.fieldChooser.discoverField.addFieldTooltip', {
-            defaultMessage: 'Add field as column',
-          })}
-        >
-          <EuiButtonIcon
-            iconType="plusInCircleFilled"
-            className={actionBtnClassName}
-            onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
-              if (ev.type === 'click') {
-                ev.currentTarget.focus();
-              }
-              ev.preventDefault();
-              ev.stopPropagation();
-              toggleDisplay(field, isSelected);
-            }}
-            data-test-subj={`fieldToggle-${field.name}`}
-            aria-label={i18n.translate('discover.fieldChooser.discoverField.addButtonAriaLabel', {
-              defaultMessage: 'Add {field} to table',
-              values: { field: field.name },
-            })}
-          />
-        </EuiToolTip>
-      );
-    } else {
-      return (
-        <EuiToolTip
-          key={`tooltip-${field.name}-${field.count || 0}-${isSelected}`}
-          delay="long"
-          content={i18n.translate('discover.fieldChooser.discoverField.removeFieldTooltip', {
-            defaultMessage: 'Remove field from table',
-          })}
-        >
-          <EuiButtonIcon
-            color="danger"
-            iconType="cross"
-            className={actionBtnClassName}
-            onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
-              if (ev.type === 'click') {
-                ev.currentTarget.focus();
-              }
-              ev.preventDefault();
-              ev.stopPropagation();
-              toggleDisplay(field, isSelected);
-            }}
-            data-test-subj={`fieldToggle-${field.name}`}
-            aria-label={i18n.translate(
-              'discover.fieldChooser.discoverField.removeButtonAriaLabel',
-              {
-                defaultMessage: 'Remove {field} from table',
-                values: { field: field.name },
-              }
-            )}
-          />
-        </EuiToolTip>
-      );
-    }
-  }
-);
+function getCommonFieldItemButtonProps({
+  field,
+  isSelected,
+  toggleDisplay,
+}: GetCommonFieldItemButtonPropsParams): {
+  field: FieldItemButtonProps<DataViewField>['field'];
+  isSelected: FieldItemButtonProps<DataViewField>['isSelected'];
+  dataTestSubj: FieldItemButtonProps<DataViewField>['dataTestSubj'];
+  buttonAddFieldToWorkspaceProps: FieldItemButtonProps<DataViewField>['buttonAddFieldToWorkspaceProps'];
+  buttonRemoveFieldFromWorkspaceProps: FieldItemButtonProps<DataViewField>['buttonRemoveFieldFromWorkspaceProps'];
+  onAddFieldToWorkspace: FieldItemButtonProps<DataViewField>['onAddFieldToWorkspace'];
+  onRemoveFieldFromWorkspace: FieldItemButtonProps<DataViewField>['onRemoveFieldFromWorkspace'];
+} {
+  const dataTestSubj = `fieldToggle-${field.name}`;
+  const handler =
+    field.name === '_source' ? undefined : (f: DataViewField) => toggleDisplay(f, isSelected);
```

Comment:
Just a note, and nothin actionable, I think we can be sure that the `_source` field won't be part of the field list given we use the field_caps API ... but out of scope of this PR